specs-go/config.go

```diff
@@ -11,7 +11,7 @@ type Spec struct {
 	// Process configures the container process.
 	Process *Process `json:"process,omitempty"`
 	// Root configures the container's root filesystem.
-	Root Root `json:"root"`
+	Root *Root `json:"root,omitempty"`
```

Comment:
Does it make sense to have a `pointer-required-if-field-must-not-be-set-on-a-platform` policy instead?

In this case: keep `*Root` and `omitempty` because it `MUST NOT` be set for Hyper-V containers